lib/shared/image_viewer.dart

```diff
@@ -113,12 +115,30 @@ class _ImageViewerState extends State<ImageViewer> with TickerProviderStateMixin
     );
   }
 
+  Future<void> getImageSize(String imageUrl, context) async {
+    retrieveImageDimensions(imageUrl);
+    Size decodedImage = await retrieveImageDimensions(imageUrl);
+
+    setState(() {
+      imagewidth = decodedImage.width;
+      imageheight = decodedImage.height;
+      maxzoomlevel = max(imagewidth, imageheight) / 128;
+      debugPrint("$imagewidth + $imageheight + $maxzoomlevel");
+      areImageDimensionsLoaded = true;
+    });
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    getImageSize(widget.url!, context);
```

Comment:
Since `getImageSize` is an async function and calls `setState()`, we should change this to be the following to be safe (we want `initState` to finish initializing before `setState` is called)

```dart
WidgetsBinding.instance.addPostFrameCallback((_){
  getImageSize(widget.url!, context);
});
```

Reply:
changes **done** 
![image](https://github.com/thunder-app/thunder/assets/108044574/3eac9e38-3310-4e20-b4ee-3ffc3758282d)